ethcore/types/src/account_diff.rs

```diff
@@ -35,9 +35,15 @@ pub enum Diff<T> where T: Eq {
 	Died(T),
 }
 
-impl<T> Diff<T> where T: Eq {
+impl<T> Diff<T> {
 	/// Construct new object with given `pre` and `post`.
-	pub fn new(pre: T, post: T) -> Self { if pre == post { Diff::Same } else { Diff::Changed(pre, post) } }
+	pub fn new(pre: T, post: T) -> Self where T: Eq {
```

Comment:
I'm not that familiar with `derive rules` for `Eq`, but looks a little strange that we have to specify in the constructor that Self (T: Eq) when we have specified it in the `derive` already!

Can somebody explain this? :)

Reply:
`derive(Debug)` means

```rust
impl<T> Debug for Diff<T> where T: Debug {
  // ...
}
```

so it is a separate `impl` and has nothing to do with this one